ql/src/java/org/apache/hadoop/hive/ql/io/orc/RecordReaderImpl.java

```diff
@@ -73,6 +76,12 @@ protected RecordReaderImpl(ReaderImpl fileReader,
     } else {
       batch = this.schema.createRowBatch();
     }
+    // Workaround for ORC not initializing TimestampColumnVector properly
+    for (int i = 0; i < batch.numCols; ++i) {
+      if (batch.cols[i] instanceof TimestampColumnVector) {
+        ((TimestampColumnVector) batch.cols[i]).setIsUTC(fileReader.isUTC());
```

Comment:
do we need the same workaround for non-orc readers like parquet?

Reply:
I think that it is only necessary for ORC readers. Note that ORC `TimestampTreeReader` sets the UTC flag in https://github.com/apache/orc/blob/29c1b37955e3423681304c6e9eb3fe7884ba57bc/java/core/src/java/org/apache/orc/impl/TreeReaderFactory.java#L1364, so the fix is necessary because `TimestampFromAnyIntegerTreeReader` and similar ORC `ConvertTreeReader` don't set the flag correctly.